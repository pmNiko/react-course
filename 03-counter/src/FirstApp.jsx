import PropTypes from "prop-types";

const FirstApp = ({ title, subtitle = 'Subtitulo de muestra' , name = 'no name' }) => {
  return (
    <>
      <h1 data-testid='test-title' >{title}</h1>

      <p>{subtitle}</p>
      
      <p>{subtitle}</p>
      
      <p>{name}</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  name: PropTypes.string
}


export default FirstApp;