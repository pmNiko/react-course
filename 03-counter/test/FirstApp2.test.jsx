import { render, screen } from '@testing-library/react';
import FirstApp from '../src/FirstApp';

describe('Pruebas en el Componente <FirstApp/>', () => { 

  const title = 'Hola soy Goku!';
  const subtitle = 'Soy un subtitulo de pruebas.';

  test('Debe hacer match con el snapshopt', () => { 
    const { container } = render(<FirstApp title={title} /> )

    expect(container).toMatchSnapshot();  
  });

  test('Debe mostrar el titulo en un h1', () => { 
    render(<FirstApp title={title} /> )

    expect( screen.getByText(title) ).toBeTruthy();
    // screen.debug();
  });


  test('Debe mostrar el subtitulo en un h1', () => { 
    render(<FirstApp title={title} /> )
    
    expect( screen.getByRole( 'heading', { level: 1 } ).innerHTML ).toContain(title);
  })
  
  test('Debe de mostrar el subtitulo enviado por props', () => { 
    render(<FirstApp title={title} subtitle={subtitle} /> )
    
    expect( screen.getAllByText(subtitle) ).toHaveLength(2); 
  })


});