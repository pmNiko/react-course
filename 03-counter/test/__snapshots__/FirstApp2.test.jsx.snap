// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Componente <FirstApp/> Debe hacer match con el snapshopt 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola soy Goku!
  </h1>
  <p>
    Subtitulo de muestra
  </p>
  <p>
    Subtitulo de muestra
  </p>
  <p>
    no name
  </p>
</div>
`;
